src/include/migraphx/op/concat.hpp

```diff
@@ -73,49 +74,96 @@ struct concat
         }
         return offsets;
     }
+
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
-        if(inputs.empty())
+        // inputs can contain 1 or more shapes (variadic).  compute_shape_op ensures there must
+        // be at least 1.
+
+        if(not std::all_of(inputs.begin(), inputs.end(), [&](const shape& s) {
+               return (s.ndim() == inputs[0].ndim() and s.type() == inputs[0].type());
+           }))
         {
-            MIGRAPHX_THROW("CONCAT: Number of input tensors should exceed 0");
+            MIGRAPHX_THROW("CONCAT: input tensors are not all the same rank or type");
         }
 
-        const auto& first_shape_lens = inputs.front().lens();
-        const auto& type             = inputs.front().type();
-        for(std::size_t l = 0; l < first_shape_lens.size(); l++)
+        if(axis >= inputs[0].ndim() or axis < 0)
+            MIGRAPHX_THROW("CONCAT: axis attribute out of range");
```

Comment:
I'm removing this check because codecov flags it; there's a test but this case is rejected by TUNE_VECTOR and this line is never reached.  Our unit tests are not so pure in this case, since they assume a dependency of one module on another.